Tests/Functional/Controller/ConnectTest.php

```diff
@@ -70,24 +56,40 @@ public function test_exporting_a_product_with_variants()
         $this->importFixturesFileOnce(__DIR__ . '/_fixtures/config_fixes.sql');
         $this->importFixturesFileOnce(__DIR__ . '/_fixtures/connect_items.sql');
 
-        $client->request('POST', 'backend/Connect/insertOrUpdateProduct', ['sourceIds' => ['2','2-123','2-124']]);
+        $sourceIds = ['2','2-123','2-124'];
+
+        $client->request('POST', 'backend/Connect/insertOrUpdateProduct', ['sourceIds' => $sourceIds]);
         $responseData = $this->handleJsonResponse($client->getResponse());
 
         $this->assertTrue($responseData['success']);
+
+        $exportedItems = self::getDbalConnection()->fetchAll(
+            "SELECT export_status FROM s_plugin_connect_items WHERE source_id in (?, ?, ?) AND export_status = 'update' ",
+            $sourceIds
+        );
+        $this->assertCount(3, $exportedItems);
     }
 
     public function test_error_handling_on_export()
     {
         /** @var TestClient $client */
         $client = self::createBackendClient();
 
+        $this->importFixturesFileOnce(__DIR__ . '/_fixtures/config_fixes.sql');
         $this->importFixturesFileOnce(__DIR__ . '/_fixtures/connect_items.sql');
 
+        $connection = self::getDbalConnection();
+
+        $connection->executeQuery('DELETE FROM s_articles_prices WHERE articledetailsID = 123');
+
         $client->request('POST', 'backend/Connect/insertOrUpdateProduct', ['sourceIds' => ['2','2-123']]);
         $responseData = $this->handleJsonResponse($client->getResponse());
 
         $this->assertFalse($responseData['success']);
-        $this->assertCount(2, $responseData['messages']['price']);
+        $this->assertCount(1, $responseData['messages']['price']);
+
+        $exportedItems = $connection->fetchAll("SELECT export_status FROM s_plugin_connect_items WHERE source_id = '2' AND export_status = 'update' ");
+        $this->assertCount(1, $exportedItems);
```

Comment:
this would still be green if product with source_id 2-123 is marked 'update' despite the error
maybe it would be suficient to remove the WHERE source_id = 2

otherwise you could check the status for 2-123 and assert that it is not update or better that it is error